src/storage/store/account/link_store.rs

```diff
@@ -502,46 +479,8 @@ impl LinkStore {
             })
     }
 
-    pub fn js_get_link_adds_by_fid(mut cx: FunctionContext) -> JsResult<JsPromise> {
-        let store = get_store(&mut cx)?;
-
-        let fid = cx.argument::<JsNumber>(0)?.value(&mut cx) as u32;
-        let link_type = cx.argument::<JsString>(1).map(|s| s.value(&mut cx))?;
-        let page_options = get_page_options(&mut cx, 2)?;
-
-        let channel = cx.channel();
-        let (deferred, promise) = cx.promise();
-
-        THREAD_POOL.lock().unwrap().execute(move || {
-            let messages = Self::get_link_adds_by_fid(&store, fid, link_type, &page_options);
-
-            deferred_settle_messages(deferred, &channel, messages);
-        });
-
-        Ok(promise)
-    }
-
-    pub fn js_get_link_removes_by_fid(mut cx: FunctionContext) -> JsResult<JsPromise> {
-        let store = get_store(&mut cx)?;
-
-        let fid = cx.argument::<JsNumber>(0)?.value(&mut cx) as u32;
-        let link_type = cx.argument::<JsString>(1).map(|s| s.value(&mut cx))?;
-        let page_options = get_page_options(&mut cx, 2)?;
-
-        let channel = cx.channel();
-        let (deferred, promise) = cx.promise();
-
-        THREAD_POOL.lock().unwrap().execute(move || {
-            let messages = Self::get_link_removes_by_fid(&store, fid, link_type, &page_options);
-
-            deferred_settle_messages(deferred, &channel, messages);
-        });
-
-        Ok(promise)
-    }
-
     pub fn get_link_removes_by_fid(
-        store: &Store,
+        store: &Store<LinkStore>,
```

Comment:
Why did these change? Not a big deal, but we should keep it consistent with hubs if possible. The CastStore uses just Store

Reply:
I changed the cast store too earlier. The reason i did this is because deriving works better with the generic. I wanted to derive clone and couldn't do it with the version that existed before because of the way we embedded the store. with `Box` and `dyn`.